bundle/run/job_options.go

```diff
@@ -0,0 +1,127 @@
+package run
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/databricks/cli/bundle/config/resources"
+	"github.com/databricks/databricks-sdk-go/service/jobs"
+	flag "github.com/spf13/pflag"
+)
+
+// JobOptions defines options for running a job.
+type JobOptions struct {
+	// Task parameters are specific to the type of task.
+	dbtCommands       []string
+	jarParams         []string
+	notebookParams    map[string]string
+	pipelineParams    map[string]string
+	pythonNamedParams map[string]string
+	pythonParams      []string
+	sparkSubmitParams []string
+	sqlParams         map[string]string
+
+	// Job parameters are a map of key-value pairs that are passed to the job.
+	// If a job uses job parameters, it cannot use task parameters.
+	// Also see https://docs.databricks.com/en/workflows/jobs/settings.html#add-parameters-for-all-job-tasks.
+	jobParams map[string]string
+}
+
+func (o *JobOptions) Define(fs *flag.FlagSet) {
+	// Define task parameters flags.
+	fs.StringSliceVar(&o.dbtCommands, "dbt-commands", nil, "A list of commands to execute for jobs with DBT tasks.")
+	fs.StringSliceVar(&o.jarParams, "jar-params", nil, "A list of parameters for jobs with Spark JAR tasks.")
+	fs.StringToStringVar(&o.notebookParams, "notebook-params", nil, "A map from keys to values for jobs with notebook tasks.")
+	fs.StringToStringVar(&o.pipelineParams, "pipeline-params", nil, "A map from keys to values for jobs with pipeline tasks.")
+	fs.StringToStringVar(&o.pythonNamedParams, "python-named-params", nil, "A map from keys to values for jobs with Python wheel tasks.")
+	fs.StringSliceVar(&o.pythonParams, "python-params", nil, "A list of parameters for jobs with Python tasks.")
+	fs.StringSliceVar(&o.sparkSubmitParams, "spark-submit-params", nil, "A list of parameters for jobs with Spark submit tasks.")
+	fs.StringToStringVar(&o.sqlParams, "sql-params", nil, "A map from keys to values for jobs with SQL tasks.")
+
+	// Define job parameters flag.
+	fs.StringToStringVar(&o.jobParams, "params", nil, "comma separated k=v pairs for job parameters")
```

Comment:
What does this look like in the help output?

For e.g. notebook params, stringToString looks pretty weird:

```
--notebook-params stringToString       A map from keys to values for jobs with notebook tasks. (default [])
```

Reply:
It looks the same. We should address this in a separate change (I copied most of the code from `job.go`).

Reply:
Sounds good.